pi_top_usb_setup/__main__.py

```diff
@@ -12,18 +13,36 @@
 click_logging.basic_config(logger)
 
 
+def is_device(path: str) -> bool:
+    return path.startswith("/dev/")
+
+
+def find_mount_point(device: str) -> str:
+    try:
+        return run_command(f"findmnt -n -o TARGET {device}", timeout=5).strip()
+    except Exception as e:
+        raise Exception(f"Error finding mount point for {device}: {e}") from e
+
+
 @click.command()
 @click_logging.simple_verbosity_option(logger)
 @click.version_option()
-@click.argument("mount_point", type=click.Path(exists=True))
+@click.argument("mount_point_or_device", type=click.Path(exists=True))
 @click.option("--skip-dialog", is_flag=True)
 def main(
-    mount_point,
+    mount_point_or_device,
     skip_dialog,
 ) -> None:
-
-    if not UsbSetupStructure.is_valid_mount_point(mount_point):
-        raise Exception(f"Couldn't find a valid USB update bundle in {mount_point}")
+    mount_point = mount_point_or_device
+    if is_device(mount_point_or_device):
+        mount_point = find_mount_point(mount_point_or_device)
+        logger.info(f"Mount point for {mount_point_or_device} is {mount_point}")
+
+    try:
+        UsbSetupStructure(mount_point)
+    except Exception as e:
+        logger.error(f"{e}")
+        return
```

Comment:
this is probably fine but maybe you didn't mean to change it to stop using is_valid_mount_point?